client/my-sites/email/email-providers-comparison/in-depth/select-button.tsx

```diff
@@ -29,7 +29,11 @@ const usePlanAvailable = (
 		return true;
 	}
 
-	if ( ! canPurchaseGSuite || ! hasGSuiteSupportedDomain( [ domain ] ) ) {
+	if ( ! canPurchaseGSuite ) {
+		return false;
+	}
+
+	if ( ! domain || ! hasGSuiteSupportedDomain( [ domain ] ) ) {
```

Comment:
Isn't better to refactor this into one single function? something like: `isGoogleWorkspacePurchaseAllowedForDomain ( domain )` and under one single condition merged?

```
const isGoogleWorkspacePurchaseAllowedForDomain = ( domain: SiteDomain | undefined ): boolean {
		return ( ! domain || ! hasGSuiteSupportedDomain( [ domain ] ) ) || ! canPurchaseGSuite;
}
```

Reply:
I'm not sure. The `usePlanAvailable()` function isn't too long.